lib/model/model.go

```diff
@@ -2675,17 +2706,25 @@ func (m *model) Availability(folder string, file protocol.FileInfo, block protoc
 	m.fmut.RUnlock()
 
 	if !ok {
-		return nil
+		return nil, ErrFolderMissing
 	}
 
-	var availabilities []Availability
-	snap := fs.Snapshot()
+	snap, err := fs.Snapshot()
+	if err != nil {
+		return nil, err
+	}
 	defer snap.Release()
+
+	return m.availabilityWithInfo(cfg, snap, file, block), nil
+}
+
+func (m *model) availabilityWithInfo(cfg config.FolderConfiguration, snap *db.Snapshot, file protocol.FileInfo, block protocol.BlockInfo) []Availability {
```

Comment:
`...WithSnapshot` seems more specific (or maybe even *In*Snapshot), or is there some other info we're looking for here?

Reply:
The unspecific "info" qualifier refers to already having the folder config and snapshot instead of needing to get them from the model, and `WithConfigAndSnapshot` is a mouthful.

Reply:
Well... I'd still go with something like `snapshotAvailability` and skip mentioning the config, at least it says more that it acts on a point in time than just "with info" which is sort of saying nothing. :)